sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,434 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import logging
+
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+if daal_check_version((2024, "P", 600)):
+    from abc import ABC
+
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.linear_model import Ridge as sklearn_RidgeRegression
+    from sklearn.metrics import r2_score
+    from sklearn.utils.validation import check_is_fitted, check_X_y
+
+    from daal4py.sklearn.linear_model._ridge import _fit_ridge as daal4py_fit_ridge
+
+    if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+        from sklearn.linear_model._base import _deprecate_normalize
+
+    from onedal.linear_model import LinearRegression as onedal_RidgeRegression
+    from onedal.utils import _num_features, _num_samples
+
+    from ..._device_offload import dispatch, wrap_output_data
+    from ..._utils import PatchingConditionsChain, get_patch_message
+    from ...utils import get_namespace
+    from ...utils.validation import _assert_all_finite
+
+    def is_numeric_scalar(value):
```

Comment:
```suggestion
    def _is_numeric_scalar(value):
```
Generally, make sense maybe add this into onedal.utils._validation, could be re-usable.

Another question? doesn't `numpy.isscalar` cover this?

Reply:
Honestly I don't see any needity for this func. it is called twice only for alpha param. Make sense just use isinstance(value, (int, float)) inplace. No need for additional function at least in ridge module. If it is required in several places in sklearnex, make sense to add in onedal.utils._validation and reuse it.